docs/Developer-Guide.md

```diff
@@ -135,26 +132,4 @@ cd <path>/sql-ds-cache
 mvn clean -DskipTests package
 ```
 
```

Comment:
It might be better to add 'guava/vmem is not recommended for the current release'

Reply:
Yes, as for we have removed the rest options in User Guide, we'd better remove these guava/vmem info in Developer Guide for current minor release, right?

Reply:
we may not remove guava/vmem in current release but in guava/vmem section, tell user need to apply numa patch if they want to use it.

Reply:
Yes, it also makes sense,  although this PR is followed by the emails decision. If not remove, please help open a new PR to upload a NUMA patch for Spark-3.1.1 and update the related docs.